### PR TITLE
Sails port

### PR DESCRIPTION
**Pull request by deini**
Alright, here goes nothing guys hahaha.

Sails port, temp API working, auth with fb, relations, etc.

@malucky @alexmckenley Good luck :mag: 

Bunch of autogenerated code:

Check out the code under `api/controllers`, `api/policies`, `api/models`, `config/routes`, `config/express`, `config/policies` ohh and  `/public` added some small temp stuff to the angular app for fb logins.

### CODE REVIEW

File: api/controllers/AuthController.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+var jwt = require('jwt-simple'),
+    moment = require('moment'),
+    request = require('request'),
+    qs = require('querystring');
+
+
+module.exports = {
+
+  me: function(req, res, next) {
```
Comment (malucky) [inline api/controllers/AuthController.js:13]: this is just temporary right?

File: api/models/Playlist.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      type: 'string',
+      defaultsTo: 'public'
+    },
+
+    owner: {
+      model: 'user'
+    },
+    songs: {
+      collection: 'song',
+      via: 'playlists'
```
Comment (malucky) [inline api/models/Playlist.js:22]: what's `via`?

File: api/policies/isSelf.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+/**
+ * isPlaylistOwner
```
Comment (malucky) [inline api/policies/isSelf.js]: copy pasted comment. how is difference from `isPlaylistOwner`?

File: config/policies.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+ *
+ * For more information on how policies work, see:
+ * http://sailsjs.org/#/documentation/concepts/Policies
+ *
+ * For more information on configuring policies, check out:
+ * http://sailsjs.org/#/documentation/reference/sails.config/sails.config.policies.html
+ */
+
+
+module.exports.policies = {
```
Comment (malucky) [inline config/policies.js:20]: this part of sails is pretty cool

File: README.md [hunk 5]
```diff
@@ -23,4 +23,4 @@ Music Providers:
 4. run `npm install -g gulp`
 5. run `gulp`
 6. run `fig up`
-7. **Linux:** Navigate to [http://localhost:3000](http://localhost:3000) **OSX:** Navigate to your boot2docker vm ip:3000 (get the ip with `boot2docker ip`)
+7. **Linux:** Navigate to [http://localhost:3000](http://localhost:3000) **OSX:** Navigate to [http://dev.ammo.io:3000](http://dev.ammo.io:3000)
```
Comment (malucky) [inline README.md:26]: this is great! where in the code did you make this work?

**Comment by malucky** [comment]
LGTM, had a few questions on how it works

File: api/controllers/AuthController.js (hunk 1 above)
Comment (deini) [inline api/controllers/AuthController.js:13]: Not really, this is the endpoint that return the current user. `/api/v1/me` will return your user. However, we can totally change the endpoint name :P

File: api/models/Playlist.js (hunk 2 above)
Comment (deini) [inline api/models/Playlist.js:22]: This is how I'm declaring the relationship. In this case Playlists has songs and I'm basically saying that songs is a bunch of `song` (declared in `/models/Song.js`) and since it's a many-to-many relationship we are just defining how everything will look in our tables. 

So back to your question ... `via` means how we are linking them, if you go to `/models/Song.js` you will see the opposite `via: 'songs'`. I don't know if I explained myself :dancer:

File: config/policies.js (hunk 4 above)
Comment (deini) [inline config/policies.js:20]: Indeed :whale:

File: README.md (hunk 5 above)
Comment (deini) [inline README.md:26]: So this was pretty annoying actually. When I was coding Fb authentication I was having some issues. The main issue was that when you create an app in fb you give it a legit IP, URL, or `localhost`. However .... when we are developing in our docker containers in OSX we are not really hitting `localhost` we have to go to our `boot2docker ip` which is a local IP facebook won't allow us to use. So.... what I came up for now as a temp fix is just to use the `boot2docker ip` and set up a sub domain. 

When we are developing in Linux you are running docker locally so you can actually just use `localhost` without a problem.

**tl;dr** No code was changed, just our domain settings.

**Comment by deini** [comment]
@malucky  4544505

File: api/controllers/AuthController.js (hunk 1 above)
Comment (malucky) [inline api/controllers/AuthController.js:13]: oh i see cool

File: api/models/Playlist.js (hunk 2 above)
Comment (malucky) [inline api/models/Playlist.js:22]: oh ok cool cool

File: README.md (hunk 5 above)
Comment (malucky) [inline README.md:26]: i see cool

**Comment by malucky** [comment]
:+1:

**Comment by alexmckenley** [comment]
This is awesome! I think at some point we'll probably want to delete some of the generic sails comments, but for now the'll probably prove super useful, and I think we should leave them. Other than a few white space issues this PR looks good to me. We'll get the .jscsrc added in soon enough, and fix them as we go. I think you should go ahead and MERGE :+1:
